test/e2e/componentreadiness/regressiontracker/regressiontracker_test.go

```diff
@@ -0,0 +1,178 @@
+package regressiontracker
+
+import (
+	"database/sql"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/lib/pq"
+	"github.com/openshift/sippy/pkg/api/componentreadiness"
+	"github.com/openshift/sippy/pkg/apis/api/componentreport"
+	"github.com/openshift/sippy/pkg/db"
+	"github.com/openshift/sippy/pkg/flags"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func cleanupAllRegressions(dbc *db.DB) {
+	// Delete all test regressions in the e2e postgres db.
+	res := dbc.DB.Where("1 = 1").Delete(&componentreport.TestRegression{})
+	if res.Error != nil {
+		log.Errorf("error deleting test regressions: %v", res.Error)
+	}
+}
+
+func Test_RegressionTracker(t *testing.T) {
+	require.NotEqual(t, "", os.Getenv("SIPPY_E2E_DSN"),
+		"SIPPY_E2E_DSN environment variable not set")
+
+	psqlFlags := flags.NewPostgresDatabaseFlags(os.Getenv("SIPPY_E2E_DSN"))
```

Comment:
What's the reason behind using a different environment variable here? It'd be better if NewPostgresDatabaseFlags didn't take arguments.

Reply:
Maybe I can find a way to set the flag opt programatically without modifying the constructor.

Reply:
A simple flags.DSN did the trick, new change pushed

Reply:
That's a good point, it's no code reuse at all really, I'll flip it to no use of flags.